src/include/homestore/btree/btree.hpp

```diff
@@ -95,29 +120,35 @@ class Btree {
 
     nlohmann::json get_metrics_in_json(bool updated = true);
     bnodeid_t root_node_id() const;
+
     uint64_t root_link_version() const;
     void set_root_node_info(const BtreeLinkInfo& info);
 
     // static void set_io_flip();
     // static void set_error_flip();
+#ifdef _PRERELEASE
+    void set_flip_point(std::string flip) { m_flips.set_flip(flip); }
+    void set_flips(std::vector< std::string > flips) {
+        for (const auto& flip : flips) {
+            set_flip_point(flip);
+        }
+    }
+    std::string flip_list() const { return m_flips.list(); }
+#endif
 
 protected:
     /////////////////////////// Methods the underlying store is expected to handle ///////////////////////////
     virtual BtreeNodePtr alloc_node(bool is_leaf) = 0;
-    virtual BtreeNode* init_node(uint8_t* node_buf, uint32_t node_ctx_size, bnodeid_t id, bool init_buf,
-                                 bool is_leaf) const;
+    virtual BtreeNode* init_node(uint8_t* node_buf, bnodeid_t id, bool init_buf, bool is_leaf) const;
     virtual btree_status_t read_node_impl(bnodeid_t id, BtreeNodePtr& node) const = 0;
     virtual btree_status_t write_node_impl(const BtreeNodePtr& node, void* context) = 0;
     virtual btree_status_t refresh_node(const BtreeNodePtr& node, bool for_read_modify_write, void* context) const = 0;
     virtual void free_node_impl(const BtreeNodePtr& node, void* context) = 0;
-    virtual btree_status_t prepare_node_txn(const BtreeNodePtr& parent_node, const BtreeNodePtr& child_node,
-                                            void* context) = 0;
-    virtual btree_status_t transact_write_nodes(const folly::small_vector< BtreeNodePtr, 3 >& new_nodes,
-                                                const BtreeNodePtr& child_node, const BtreeNodePtr& parent_node,
-                                                void* context) = 0;
-
+    virtual btree_status_t transact_nodes(const BtreeNodeList& new_nodes, const BtreeNodeList& freed_nodes,
+                                          const BtreeNodePtr& left_child_node, const BtreeNodePtr& parent_node,
+                                          void* context) = 0;
```

Comment:
With the latest changes, we no need prepare a transaction and then do transaction, but simply call transact_nodes and pass all operations/nodes which needs to be atomically updated.